src/jit/emitxarch.cpp

```diff
@@ -5564,22 +5447,30 @@ void emitter::emitIns_I_ARX(
 }
 
 void emitter::emitIns_R_ARX(
-    instruction ins, emitAttr attr, regNumber ireg, regNumber base, regNumber index, unsigned mul, int disp)
+    instruction ins, emitAttr attr, regNumber reg, regNumber base, regNumber index, unsigned scale, int disp)
 {
-    assert((CodeGen::instIsFP(ins) == false) && (EA_SIZE(attr) <= EA_8BYTE) && (ireg != REG_NA));
-    noway_assert(emitVerifyEncodable(ins, EA_SIZE(attr), ireg));
+    assert(!CodeGen::instIsFP(ins) && (EA_SIZE(attr) <= EA_32BYTE) && (reg != REG_NA));
+    noway_assert(emitVerifyEncodable(ins, EA_SIZE(attr), reg));
+
+    if ((ins == INS_lea) && (reg == base) && (index == REG_NA) && (disp == 0))
+    {
+        // Maybe the emitter is not the common place for this optimization, but it's a better choke point
```

Comment:
I'll run some diffs without code to see what happens. I was surprised when I first encountered this, I have no idea how such a LEA could be produced to begin with. The address mode generation code is pretty hairy but it can't be that stupid.

Reply:
> The address mode generation code is pretty hairy but it can't be that stupid.

Oh, yes, it really is that stupid. It does really generate `LEA(b+0)` from `ADD b, 0`. Well, the fact that `ADD b, 0` reaches LIR may seem stupid as well but there's an excuse for that - there's a field sequence attached to 0.

Is it worth creating an issue to keep track of this?

Reply:
> Is it worth creating an issue to keep track of this?

Thanks, I think this behavior needs an issue.

Reply:
Created #27092